src/DotVVM.Framework/ResourceManagement/ResourcesRenderer.cs

```diff
@@ -18,12 +18,22 @@ public static void RenderResourceCached(this NamedResource resource, IHtmlWriter
             writer.WriteUnencodedText(resource.GetRenderedTextCached(context));
         }
 
-        public static string GetRenderedTextCached(this NamedResource resource, IDotvvmRequestContext context) => 
+        public static string GetRenderedTextCached(this NamedResource resource, IDotvvmRequestContext context) =>
             // dont use cache when debug, so the resource can be refreshed when file is changed
             context.Configuration.Debug ?
             RenderToString(resource, context) :
             renderedCache.GetValue(resource.Resource, _ => RenderToString(resource, context));
 
+        public static void RenderPreload(this IPreloadResource resource, string resourceName, IHtmlWriter writer, IDotvvmRequestContext context)
```

Comment:
I'd rather let the implementation of IPreloadResource decide what does it want to render into the header. IMO, it will be more powerful and more systematic (almost same signature as Render).